### PR TITLE
compatibility with newest gensim and sklearn

### PR DESCRIPTION
**Pull request by GTmac**
Hi Bryan,

I am trying to make DeepWalk compatible with the newest versions of gensim and sklearn. I test the code below with gensim 3.1.0 and sklearn 0.19.1, and it can almost perfectly reproduce the original result on BlogCatalog. The changes I made are the following:
1. Enforce the default parameters for gensim word2vec by setting sg=1 (use Skip-gram) and hs=1 (use hierarchical softmax).
2. convert int random walks into strings, since newer versions of gensim only support arrays of strings as input.
3. Replace combine_files_iter with a class named "WalksCorpus", since newer version of gensim does not support generator any more (for reasons I don't understand).
4. In example_graph/scoring.py, binarize node labels before classification for compatibility with modern sklearn.
5. I also remove the build_vocab() function in deepwalk/SkipGram.py, since it is not compatible with the current gensim. I am not sure if this is an appropriate change.
6. Update requirements.txt accordingly.

Please take a look, thank you so much!

### CODE REVIEW

**Comment by phanein** [comment]
Thanks @GTmac !  I'm merging this, but have some minor comments:

1. Can you also update the README to indicate that this is a new version?  My one concern with using strings has always been scalability for the larger graphs - let's make it clear that to completely reproduce the paper, one will probably have to install the old version of everything.
2. Finally as an open question - how does this effect parallelization?

**Comment by GTmac** [comment]
Thanks Bryan :)
1. I have upated the README file accordingly.
2. It seems to me that training word2vec is much more time consuming than random walking, so I would say this would not affect parallelization (a lot). But I have not empirically compared the running time of old DeepWalk and new DeepWalk yet.
